frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,484 @@
+  InstallPlanModel,
+  PackageManifestModel,
+  SubscriptionModel,
+} from '../models';
+import {
+  CRDDescription,
+  ClusterServiceVersionKind,
+  SubscriptionKind,
+  InstallPlanKind,
+  PackageManifestKind,
+} from '../types';
+import {
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared/src/components/status/icons';
+import { InstallPlanPreview } from './install-plan';
+import { RedExclamationCircleBadge } from '@console/shared/src/components/badges/RedExclamationCircleBadge';
+
+const getCRDFromObject = (obj: ClusterServiceVersionKind | InstallPlanKind) => {
+  return obj?.metadata?.annotations?.['operatorframework.io/initialization-resource'];
+};
+
+const isCRDAvailable = (obj: ClusterServiceVersionKind | InstallPlanKind) =>
+  typeof getCRDFromObject(obj) !== 'undefined';
+
+const installPageResources = (namespace: string) => [
+  {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    namespaced: true,
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('csvName')}`,
+    isList: true,
+    namespace,
+    prop: 'clusterServiceVersions',
+  },
+  {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('csvName')}`,
+    isList: true,
+    namespace: 'openshift-operators',
+    prop: 'allClusterServiceVersions',
+  },
+  {
+    kind: referenceForModel(SubscriptionModel),
+    namespaced: false,
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('pkg')}`,
+    isList: true,
+    optional: true,
+    prop: 'subscription',
+  },
+  {
+    kind: referenceForModel(InstallPlanModel),
+    prop: 'installPlan',
+    namespaced: false,
+    isList: true,
+    optional: true,
+  },
+  {
+    isList: true,
+    kind: referenceForModel(PackageManifestModel),
+    namespace: new URLSearchParams(window.location.search).get('catalogNamespace'),
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('pkg')}`,
+    selector: {
+      matchLabels: {
+        catalog: new URLSearchParams(window.location.search).get('catalog'),
+      },
+    },
+    prop: 'packageManifest',
+  },
+];
+
+const ViewInstalledOperatorsButton: React.FC<InstallPageMessageButtonProps> = ({
+  namespace,
+  className,
+}) => (
+  <Button
+    className={className}
+    data-test-id="view-installed-operators-btn"
+    variant="link"
+    component="a"
+    href={`/k8s/ns/${namespace}/${referenceForModel(ClusterServiceVersionModel)}`}
+  >
+    View Installed Operators
+  </Button>
+);
+
+const InstallFailedMessage: React.FC<InstallPageMessageProps> = ({ namespace, csvName, obj }) => {
+  const crdMessage = isCRDAvailable(obj)
+    ? 'The required custom resource will need to be created in the operator once resolved.'
+    : '';
+  const installMessage = `The operator did not install successfully. ${crdMessage}`;
```

Comment:
Don't the string this way. This will make things harder if we ever translate console. Instead do something like this below.

```tsx
<span>The operator did not install successfully. {crdMessage}</span>